metacat-common-server/src/main/java/com/netflix/metacat/common/server/util/DataSourceManager.java

```diff
@@ -30,15 +29,28 @@
 /**
  * Data source manager.
  */
+@Slf4j
 public final class DataSourceManager {
+
     private static final String JDO_PREFIX = "javax.jdo.option.";
     private static DataSourceManager instance = new DataSourceManager();
     private Map<String, DataSource> dataSources = Maps.newConcurrentMap();
+    private Environment environment;
 
     private DataSourceManager() {
     }
 
     /**
+     * Set the environment instance.
+     *
+     * @param environment environment
+     */
+    @Autowired
+    public void setEnvironment(final Environment environment) {
```

Comment:
I think there is a Spring interface `EnvironmentAware` or something like that which you can implement to have spring inject the Environment. Still think we need to sit down and fully redesign this class so it's not doing setter injection.